synapse/types.py

```diff
@@ -751,3 +751,32 @@ def get_verify_key_from_cross_signing_key(key_info):
     # and return that one key
     for key_id, key_data in keys.items():
         return (key_id, decode_verify_key_bytes(key_id, decode_base64(key_data)))
+
+
+@attr.s(frozen=True, slots=True)
+class User:
+    """Represents a user object.
+
+    Attributes:
+        user:  ID of the user.
+        admin:  True if the user is an admin.
+        app_service:  Application service that created this user.
+        consent_server_notice_sent:  Version of policy documents the user has been sent.
+        consent_version:  Version of policy documents the user has consented to.
+        creation_ts:  Creation timestamp of the user.
+        deactivated:  True if the user has been deactivated.
+        is_guest:  True if the user is a guest user.
+        shadow_banned:  True if the user has been shadow-banned.
+        user_type:  User type (None for normal user, 'support' and 'bot' other options).
+    """
+
+    user = attr.ib(type="UserID")
+    admin = attr.ib(type=bool, default=False, converter=bool)
```

Comment:
why do we need `converter`s on this and the others below? If we're trying to assign something other than a bool then that suggests something wrong in the caller?

Reply:
I added those since the database record from `_store.get_user_by_id` seemed to not return bools. Agreed this is not the place to do such things, removing.